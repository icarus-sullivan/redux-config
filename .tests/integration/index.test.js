import { createStore, applyMiddleware } from 'redux';
import { constants, actions, reducers } from '../../lib';

describe('Integration', () => {
  it('works with redux', () => {
    expect.assertions(3);
    const Constants = constants([
      {
        invocation: 'async',
        namespace: 'comment',
        verbs: ['post'],
      },
      {
        namespace: 'comment',
        verbs: ['view'],
      },
    ]);

    const reducer = reducers([
      {
        type: Constants.COMMENT_POST.REQUESTED,
        path: 'comments.loading',
        fn: () => true,
      },
      {
        type: Constants.COMMENT_POST.SUCCEEDED,
        path: 'comments.data',
        fn: (state = [], post) => [...state, ...post],
      },
      {
        type: Constants.COMMENT_POST.DONE,
        path: 'comments.loading',
        fn: () => false,
      },
    ]);

    const actionCreator = actions({
      postComment: {
        invocation: 'async',
        type: Constants.COMMENT_POST.DEFAULT,
        fn: async (msg) => {
          return [
            {
              id: '111-111-111-111',
            },
          ];
        },
      },
    });

    const logger = ({ getState }) => (next) => (action) => {
      const returnVal = next(action);

      expect([getState(), action]).toMatchSnapshot();
      return returnVal;
    };

    const store = createStore(reducer, {}, applyMiddleware(logger));

    const actions = actionCreator(store.dispatch);

    actions.postComment('hello world');
  });
});
