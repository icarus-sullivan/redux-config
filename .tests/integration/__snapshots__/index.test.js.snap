// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration works with redux 1`] = `
Array [
  Object {
    "comments": Object {
      "loading": true,
    },
  },
  Object {
    "payload": "hello world",
    "type": "@COMMENT/POST_REQUESTED",
  },
]
`;

exports[`Integration works with redux 2`] = `
Array [
  Object {
    "comments": Object {
      "data": Array [
        Object {
          "id": "111-111-111-111",
        },
      ],
      "loading": true,
    },
  },
  Object {
    "payload": Array [
      Object {
        "id": "111-111-111-111",
      },
    ],
    "type": "@COMMENT/POST_SUCCEEDED",
  },
]
`;

exports[`Integration works with redux 3`] = `
Array [
  Object {
    "comments": Object {
      "data": Array [
        Object {
          "id": "111-111-111-111",
        },
      ],
      "loading": false,
    },
  },
  Object {
    "type": "@COMMENT/POST_DONE",
  },
]
`;
